### ISSUE TITLE
The total of genes in GeneRatio can be higher than the initial number of genes

### ISSUE THREAD

**Issue by dalloliogm** [GuangchuangYu/clusterProfiler#8]
More than an issue, this is just a question.

Let's calculate the enrichment of three genes: 1, 100 and 1000

```
> summary(groupGO(as.character(c(1,100,1000))))
                  ID                               Description Count GeneRatio     geneID
GO:0016020 GO:0016020                                  membrane     2       2/5   100/1000
GO:0005576 GO:0005576                      extracellular region     3       3/5 1/100/1000
GO:0005581 GO:0005581                           collagen trimer     0       0/5           
GO:0005623 GO:0005623                                      cell     2       2/5   100/1000
GO:0009295 GO:0009295                                  nucleoid     0       0/5           
GO:0019012 GO:0019012                                    virion     0       0/5           
GO:0030054 GO:0030054                             cell junction     2       2/5   100/1000
GO:0031012 GO:0031012                      extracellular matrix     0       0/5           
```

In the GeneRatio column, the output shows that there are 2 out of 5 genes matching the term "membrane". However, the input vector contained only three Entrez Ids (1, 100 and 1000). Shouldn't the total be 3 genes instead of 5? 

I imagine that this can be due to one-to-many relationships between Entrez Ids and GeneOntology. For example, Entrez 1 may correspond to more than one gene in GO. However, how can I check this?

**Comment by GuangchuangYu** [comment]
Good catch. That's a bug due to a typo. 

It will be fixed in version >=2.1.7

**Comment by YichaoCaiDBS** [comment]
Sorry to bring up this issue again. 

I found that the total number of genes in the results does not match what is input into the enrichGO(). In GeneRatio, the total number of genes are fewer than the input vector.

For example, if we take the example used in the GO classfication from the manual (https://yulab-smu.github.io/clusterProfiler-book/chapter5.html). We can do this:
```
library(clusterProfiler)
data(geneList, package="DOSE")
gene <- names(geneList)[abs(geneList) > 2]
length(unique(gene)) ## 207 genes

ego <- enrichGO(gene          = gene,
                universe      = names(geneList),
                OrgDb         = org.Hs.eg.db,
                ont           = "CC",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05,
        readable      = TRUE)
head(ego)
```
The result would look like this:
```
ID                              Description GeneRatio   BgRatio       pvalue     p.adjust       qvalue
GO:0005819 GO:0005819                                  spindle    25/202 262/11812 2.569986e-12 8.146855e-10 7.520590e-10
GO:0005876 GO:0005876                      spindle microtubule    12/202  45/11812 7.912518e-12 1.254134e-09 1.157726e-09
GO:0000779 GO:0000779 condensed chromosome, centromeric region    15/202  91/11812 3.254384e-11 3.438799e-09 3.174452e-09
GO:0072686 GO:0072686                          mitotic spindle    14/202  84/11812 1.292642e-10 9.903693e-09 9.142376e-09
GO:0000775 GO:0000775           chromosome, centromeric region    18/202 156/11812 1.562097e-10 9.903693e-09 9.142376e-09
GO:0000776 GO:0000776                              kinetochore    15/202 106/11812 3.073022e-10 1.623580e-08 1.498772e-08
                                                                                                                                                        geneID
GO:0005819 CDCA8/CDC20/KIF23/CENPE/ASPM/DLGAP5/SKA1/NUSAP1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF18A/BIRC5/KIF11/TTK/AURKB/PRC1/KIFC1/KIF18B/KIF20A/AURKA/CCNB1/KIF4A
GO:0005876                                                                             CENPE/SKA1/NUSAP1/CDK1/KIF18A/BIRC5/KIF11/AURKB/PRC1/KIF18B/AURKA/KIF4A
GO:0000779                                                            CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/CDT1/BIRC5/NCAPG/AURKB/AURKA/CCNB1
GO:0072686                                                                KIF23/CENPE/ASPM/NUSAP1/TPX2/TACC3/CDK1/MAD2L1/KIF18A/KIF11/AURKB/KIFC1/KIF18B/AURKA
GO:0000775                                           CDCA8/CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/NCAPG/AURKB/AURKA/CCNB1
GO:0000776                                                             CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/AURKB/CCNB1
           Count
GO:0005819    25
GO:0005876    12
GO:0000779    15
GO:0072686    14
GO:0000775    18
GO:0000776    15
```

I am using clusterProfiler 3.10. Is this still a bug? If not, why are we getting inconsistent gene numbers?

Thank you!

PS: This is my sessionInfo():
```
R version 3.5.1 (2018-07-02)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS: /home/yichao/anaconda3/lib/R/lib/libRblas.so
LAPACK: /home/yichao/anaconda3/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_SG.UTF-8       LC_NUMERIC=C               LC_TIME=en_SG.UTF-8        LC_COLLATE=en_SG.UTF-8    
 [5] LC_MONETARY=en_SG.UTF-8    LC_MESSAGES=en_SG.UTF-8    LC_PAPER=en_SG.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_SG.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] org.Hs.eg.db_3.7.0     AnnotationDbi_1.44.0   IRanges_2.16.0         S4Vectors_0.20.1       Biobase_2.42.0        
[6] BiocGenerics_0.28.0    clusterProfiler_3.10.1

loaded via a namespace (and not attached):
 [1] viridis_0.5.1       httr_1.4.1          tidyr_1.0.0         tidygraph_1.1.2     bit64_0.9-7         jsonlite_1.6       
 [7] viridisLite_0.3.0   splines_3.5.1       ggraph_2.0.0        assertthat_0.2.1    DO.db_2.9           BiocManager_1.30.10
[13] rvcheck_0.1.7       triebeard_0.3.0     urltools_1.7.3      blob_1.2.0          yaml_2.2.0          progress_1.2.2     
[19] ggrepel_0.8.1       pillar_1.4.2        RSQLite_2.1.2       backports_1.1.5     lattice_0.20-38     glue_1.3.1         
[25] digest_0.6.23       RColorBrewer_1.1-2  ggsignif_0.6.0      polyclip_1.10-0     qvalue_2.14.1       colorspace_1.4-1   
[31] cowplot_1.0.0       Matrix_1.2-17       plyr_1.8.4          pkgconfig_2.0.3     purrr_0.3.3         GO.db_3.7.0        
[37] scales_1.1.0        ggplotify_0.0.4     europepmc_0.3       tweenr_1.0.1        enrichplot_1.2.0    BiocParallel_1.16.6
[43] ggforce_0.3.1       tibble_2.1.3        farver_2.0.1        ggplot2_3.2.1       UpSetR_1.4.0        lazyeval_0.2.2     
[49] magrittr_1.5        crayon_1.3.4        memoise_1.1.0       DOSE_3.8.0          MASS_7.3-51.4       xml2_1.2.2         
[55] prettyunits_1.0.2   tools_3.5.1         data.table_1.12.6   hms_0.5.2           lifecycle_0.1.0     stringr_1.4.0      
[61] munsell_0.5.0       compiler_3.5.1      gridGraphics_0.4-1  rlang_0.4.2         ggridges_0.5.1      grid_3.5.1         
[67] rstudioapi_0.10     igraph_1.2.4.2      gtable_0.3.0        DBI_1.0.0           reshape2_1.4.3      graphlayouts_0.5.0 
[73] R6_2.4.1            gridExtra_2.3       dplyr_0.8.3         bit_1.1-14          zeallot_0.1.0       fastmatch_1.1-0    
[79] fgsea_1.8.0         GOSemSim_2.8.0      stringi_1.4.3       Rcpp_1.0.3          vctrs_0.2.0         tidyselect_0.2.5   
```
